pkgs/development/tools/kubedock/default.nix

```diff
@@ -0,0 +1,34 @@
+{ lib, buildGoModule, fetchFromGitHub }:
+
+let
+  gitref = "432177c8033824f2d432fc465b98b59d21cfeefb";
+in
+buildGoModule rec {
+  pname = "kubedock";
+  version = "0.14.1";
+
+  src = fetchFromGitHub {
+    owner = "joyrex2001";
+    repo = "kubedock";
+    rev = gitref;
+    hash = "sha256-xjt/RqvkOHTSTZwNgxmlUeYndPW9jwUE1knLyNXDxaE=";
+  };
+
+  vendorHash = "sha256-4X4v40c4J5SPwj53Zk+j+A9k0RkHGwqz128E+FnLr1c=";
+
+  ldflags = [
+    "-s"
+    "-w"
+    "-X github.com/joyrex2001/kubedock/internal/config.Build=${gitref}"
```

Comment:
Upstream seems to have version tags, is the gitref strictly neccesary?

Reply:
Not sure what you mean? this is what I get from running it on docker:
```
$ docker run --rm joyrex2001/kubedock version
-------------------------------------------------
kubedock
-------------------------------------------------
version: 0.14.1-7-g06b1305
date:    20231118-124310
build:   06b1305d9ffc318eca4e8f9f7840b38f71a6692e
-------------------------------------------------
```

That build value points to commit https://github.com/joyrex2001/kubedock/commit/06b1305d9ffc318eca4e8f9f7840b38f71a6692e

The build upstream sets this in https://github.com/joyrex2001/kubedock/blob/5aad204a72fcd45b1540ed443d7cfff2fed9ece0/Makefile#L2

Without this it prints `<undef>`

Reply:
Would `config.Build=refs/tags/${version}` work? The git ref makes it so r-ryantm can't do its job on this package

Reply:
Ah I see. Unfortunately that doesn't work...
I see a few options here, none of them ideal:
1. Leave it as it is and forgo r-ryantm
2. Use `leaveDotGit` ([people say it has issues](https://discourse.nixos.org/t/keep-git-folder-in-when-fetching-a-git-repo/8590) though)
3. Leave `config.Build` undefined

Any other ideas? I'd go for option 3 here.

Reply:
Option 1 is also fine, if you add adequate reasoning in comments.

Reply:
I went with option 3 in https://github.com/NixOS/nixpkgs/pull/268352/commits/6e51d84a14afc9597646a92be79bbe8ba7607c45